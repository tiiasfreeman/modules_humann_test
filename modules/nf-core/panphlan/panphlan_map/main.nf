process PANPHLAN_MAP {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-e3818ba00c0f15bcda92c714b4fa500c626067fe:cae224e6965f9abd8b08407231aff3bba632f1de-0':
        'biocontainers/mulled-v2-e3818ba00c0f15bcda92c714b4fa500c626067fe:cae224e6965f9abd8b08407231aff3bba632f1de-0' }"

    input:
    path(pangenome)
    path(indexes)
    tuple val(meta), path(sample_sequence)
    val(species_name)

    output:
    tuple val(meta), path("${species_name}_map")    , emit: mapping_dir
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    panphlan_map.py \\
        -p $pangenome \\
        --indexes ${indexes}/${species_name} \\
        -i ${sample_sequence} \\
        -o ${meta.id}_${species_name}.tsv \\
        ${args}

    mkdir ${species_name}_map
    mv ${meta.id}_${species_name}.tsv ${species_name}_map

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        panphlan: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        panphlan: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
}
