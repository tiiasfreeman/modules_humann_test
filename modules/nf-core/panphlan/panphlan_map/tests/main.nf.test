nextflow_process {

    name "Test Process PANPHLAN_MAP"
    script "modules/nf-core/panphlan/panphlan_map/main.nf"
    process "PANPHLAN_MAP"

    setup {
        run("PANPHLAN_DOWNLOADPANGENOME") {
            script "modules/nf-core/panphlan/downloadpangenome/main.nf"
                process {
                    """
                    input[0] = "Bacteroides_fragilis"
                    """
                }
            }
        }

    test("Bacteroides fragilis sample fastq - single_end") {

        when {
            process {
                """
                input[0] = PANPHLAN_DOWNLOADPANGENOME.out.pangenome
                input[1] = PANPHLAN_DOWNLOADPANGENOME.out.indexes
                input[2] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test2_1.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test2_2.fastq.gz", checkIfExists:true)
                ])
                input[3] = "Bacteroides_fragilis"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }

    test("Bacteroides fragilis sample fastq - single_end '- stub' ") {

        options "-stub"

        when {
            process {
                """
                input[0] = PANPHLAN_DOWNLOADPANGENOME.out.pangenome
                input[1] = PANPHLAN_DOWNLOADPANGENOME.out.indexes
                input[2] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_1.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test1_2.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test2_1.fastq.gz", checkIfExists:true),
                        file("https://github.com/nf-core/test-datasets/raw/modules/data/genomics/prokaryotes/bacteroides_fragilis/illumina/fastq/test2_2.fastq.gz", checkIfExists:true)
                ])
                input[3] = "Bacteroides_fragilis"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}