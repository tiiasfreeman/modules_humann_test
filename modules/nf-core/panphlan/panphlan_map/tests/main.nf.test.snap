{
    "Bacteroides fragilis sample fastq - single_end '- stub' ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_Bacteroides_fragilis.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,d9e8b34830d477d673900d100ea62420"
                ],
                "mapping_dir": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_Bacteroides_fragilis.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d9e8b34830d477d673900d100ea62420"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T19:03:03.872836089"
    },
    "Bacteroides fragilis sample fastq - single_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_Bacteroides_fragilis.tsv:md5,6ac420a56792567619a08bc1d054706a"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,d9e8b34830d477d673900d100ea62420"
                ],
                "mapping_dir": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_Bacteroides_fragilis.tsv:md5,6ac420a56792567619a08bc1d054706a"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d9e8b34830d477d673900d100ea62420"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-10T19:02:46.840544315"
    }
}